internal/announcements/announce.go

```diff
@@ -0,0 +1,110 @@
+package announcements
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Event struct {
+	Type string
+	Data map[string]interface{}
+}
+
+func NewEvent(eventType string, initialData ...map[string]interface{}) *Event {
+	var data map[string]interface{}
+	if len(initialData) > 0 {
+		data = initialData[0] // Use the first map provided if any.
+	} else {
+		data = make(map[string]interface{})
+	}
+
+	return &Event{
+		Type: eventType,
+		Data: data,
+	}
+}
+
+func (e *Event) Set(key string, value interface{}) {
+	e.Data[key] = value
+}
+
+func (e *Event) Get(key string) (interface{}, error) {
+	value, exists := e.Data[key]
+	if !exists {
+		return nil, fmt.Errorf("key '%s' does not exist in event of type '%s'", key, e.Type)
+	}
+	return value, nil
+}
+
+type Announcement interface {
+	Announce(event *Event)
+	Register(monitor Monitors)
+}
+
+type NoopAnnouncement struct{}
+
+type DstAnnouncement struct {
+	announcements []Event
+	monitors      []Monitors
+	mutex         sync.Mutex // Mutex for thread safety
```

Comment:
Usually a better practice to have the mutex the start of the struct. 
As the struct need not calculate the offset every time to find the mutex variable inside the struct.